api/serializers.py

```diff
@@ -11,9 +11,35 @@
 
 
 class HuntSerializer(serializers.ModelSerializer):
+    num_solved = serializers.SerializerMethodField()
+    num_unsolved = serializers.SerializerMethodField()
+    num_unlocked = serializers.SerializerMethodField()
+    num_metas_solved = serializers.SerializerMethodField()
+
+    def get_num_solved(self, obj):
+        return obj.get_num_solved()
+
+    def get_num_unsolved(self, obj):
+        return obj.get_num_unsolved()
+
+    def get_num_unlocked(self, obj):
+        return obj.get_num_unlocked()
+
+    def get_num_metas_solved(self, obj):
+        return obj.get_num_metas_solved()
+
     class Meta:
         model = Hunt
-        fields = ("id", "name", "active", "url")
+        fields = (
+            "id",
+            "name",
+            "active",
+            "url",
+            "num_solved",
+            "num_unsolved",
+            "num_unlocked",
+            "num_metas_solved",
+        )
```

Comment:
What do you think about computing this on the front-end instead? I think I have an aesthetic preference for that vs putting all this derived data inside every Hunt object but I think it also might have a couple other advantages (always in sync with the other data you have & no need for another set of recurring API calls).

In this case I think it would work nicely as a bunch of redux selectors in `puzzlesSlice.js`

Reply:
Is there a nice way to link this up with the stats page if we do this on the front-end? Ideally, I'm aiming for this to also be displayed there, along with being polled (or at least updated) regularly with the progress chart and the other stats.

Reply:
Alternatively, we could keep the back-end functions for the stats page and move the puzzle/hunt display page's computation to a front-end one. (This does do the same thing two different ways; not sure if that's a problem...)

Reply:
I think what you have right now should be fine for the stats page, right? I think having the Hunt.num_foo() functions is fine, just not sure about the serialization alongside every other hunt. If you want endpoints that you can query for refreshing the standalone stats page (which seems reasonable--wasn't sure if you were planning on adding that) maybe something like a /hunts/:id/stats endpoint that just returns them directly?

I do have some preference for deriving them in the front-end for the main puzzles page so that they're always in sync with the data we've got.

Reply:
Sounds good. I'll look into it. (Refreshing the page is definitely part of the eventual goal, part of the next branch I was looking into; hopefully I'll be able to get there with enough time to test and spare.)